### PR TITLE
Add skudetails function to get info about sku's from bazaar

### PR DESCRIPTION
**Pull request by shayanpourvatan**

### CODE REVIEW

**Review by PHELAT — APPROVED** [review]
LGTM!

File: app/src/main/java/ir/cafebazaar/poolakeysample/MainActivity.kt [hunk 1]
```diff
@@ -84,6 +90,40 @@ class MainActivity : AppCompatActivity() {
                 payment.getSubscribedProducts(handlePurchaseQueryCallback())
             }
         }
+        setGetSkuDetailClickListener()
+    }
+
+    private fun setGetSkuDetailClickListener() {
+
+        getSkuDetailInAppButton.setOnClickListener {
+            if (paymentConnection.getState() == ConnectionState.Connected) {
```
Comment (hamidrezasahraei) [inline app/src/main/java/ir/cafebazaar/poolakeysample/MainActivity.kt:99]: In order to have better readability, maintainability, and have a smaller method I suggest moving this part to another method and call that method here.
@shayanpourvatan

File: app/src/main/java/ir/cafebazaar/poolakeysample/MainActivity.kt [hunk 2]
```diff
@@ -84,6 +90,40 @@ class MainActivity : AppCompatActivity() {
...
+                    }
+                    getSkuDetailsFailed {
+                        toast(R.string.general_query_get_sku_detail_failed_message)
+                    }
+                }
+            }
+        }
+
+        getSkuDetailSubscriptionButton.setOnClickListener {
+            if (paymentConnection.getState() == ConnectionState.Connected) {
```
Comment (hamidrezasahraei) [inline app/src/main/java/ir/cafebazaar/poolakeysample/MainActivity.kt:114]: In order to have better readability, maintainability, and have a smaller method I suggest moving this part to another method and call that method here.
@shayanpourvatan

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -11,6 +11,8 @@
     <string name="general_service_connection_failed_text">Service: Failed to Connect</string>
     <string name="general_query_purchase_text">Check if user purchased this item</string>
     <string name="general_query_subscription_text">Check if user subscribed this item</string>
+    <string name="general_get_sku_detail_inapp_text">Get Sku detail of inApp item</string>
```
Comment (hamidrezasahraei) [inline app/src/main/res/values/strings.xml:14]: We have a typo here, change `inApp` to `in-app`

File: poolakey/src/main/java/ir/cafebazaar/poolakey/billing/skudetail/GetSkuDetailFunction.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import ir.cafebazaar.poolakey.entity.SkuDetails
+import ir.cafebazaar.poolakey.exception.ResultNotOkayException
+import ir.cafebazaar.poolakey.takeIf
+import ir.cafebazaar.poolakey.thread.PoolakeyThread
+
+internal class GetSkuDetailFunction(
+    private val context: Context,
+    private val mainThread: PoolakeyThread<() -> Unit>
+) : BillingFunction<SkuDetailFunctionRequest> {
+    override fun function(
```
Comment (hamidrezasahraei) [inline poolakey/src/main/java/ir/cafebazaar/poolakey/billing/skudetail/GetSkuDetailFunction.kt:21]: Put a blank line here

File: poolakey/src/main/java/ir/cafebazaar/poolakey/billing/skudetail/GetSkuDetailFunction.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        billingService: IInAppBillingService,
+        request: SkuDetailFunctionRequest
+    ): Unit = with(request) {
+        try {
```
Comment (hamidrezasahraei) [inline poolakey/src/main/java/ir/cafebazaar/poolakey/billing/skudetail/GetSkuDetailFunction.kt:25]: This is a veryyyyy long method, Can you break it into smaller units?

**Review by hamidrezasahraei — CHANGES_REQUESTED** [review]

**Pull request merged**